### PR TITLE
OCD-4057 - Run validators only against Active Certificates

### PR DESCRIPTION
**Pull request by tmy1313**

### CODE REVIEW

**Review by kekey1 — DISMISSED** [review]

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/questionableactivity/ListingQuestionableActivityService.java [hunk 1]
```diff
@@ -28,6 +28,7 @@
 import gov.healthit.chpl.questionableactivity.listing.DeletedRwtResultsActivity;
 import gov.healthit.chpl.questionableactivity.listing.DeletedSurveillanceActivity;
 import gov.healthit.chpl.questionableactivity.listing.ListingActivity;
+import gov.healthit.chpl.questionableactivity.listing.NonActiveListingEdited;
```
Comment (andlar) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/questionableactivity/ListingQuestionableActivityService.java]: This looks like the wrong name

**Review by andlar — DISMISSED** [review]

**Review by kekey1 — APPROVED** [review]

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/listing/Validator.java [hunk 2]
```diff
@@ -29,11 +29,12 @@ public synchronized void validate(final CertifiedProductSearchDetails listing) {
         }
     }
 
-    public void validate(final CertifiedProductSearchDetails existingListing,
-            final CertifiedProductSearchDetails updatedListing) {
-        validate(updatedListing);
-        for (ComparisonReviewer reviewer : getComparisonReviewers()) {
-            reviewer.review(existingListing, updatedListing);
+    public void validate(CertifiedProductSearchDetails existingListing, CertifiedProductSearchDetails updatedListing) {
+        if (updatedListing.isCertificateActive()) {
```
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/listing/Validator.java:35]: I think this change is great for the 2015 listings. I just remembered I had hoped to change the 2014 validation to use the "AllowedListingValidator". Not recalling the name exactly but it's whatever we use for 2011 edition. There is a validator factory class which I believe assigns the correct validator to a listing. I was hoping to basically remove all references to 2014 edition listing validators in this PR and maybe, hopefully, delete those files.

**Review by kekey1 — CHANGES_REQUESTED** [review]

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/listing/ListingValidatorFactoryImpl.java [hunk 3]
```diff
@@ -1,70 +1,29 @@
...
 import org.springframework.stereotype.Service;
 
 import gov.healthit.chpl.domain.CertifiedProductSearchDetails;
-import gov.healthit.chpl.util.ChplProductNumberUtil;
 import gov.healthit.chpl.util.ErrorMessageUtil;
+import lombok.extern.log4j.Log4j2;
 
 @Service
+@Log4j2
 public class ListingValidatorFactoryImpl implements ListingValidatorFactory {
```
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/listing/ListingValidatorFactoryImpl.java]: Left comment in chat - can we remove ListingValidatorFactory and make this one "not an Impl"?

**Review by kekey1 — CHANGES_REQUESTED** [review]

**Review by kekey1 — DISMISSED** [review]

**Review by andlar — DISMISSED** [review]

**Review by kekey1 — DISMISSED** [review]
I think it might be worth adding a test or two that expect this exception is thrown if you mock up a listing with some null test tool IDs or something.

**Review by andlar — DISMISSED** [review]

File: chpl/chpl-api/src/main/resources/log4j2-console.xml [hunk 4]
```diff
@@ -15,6 +15,19 @@
...
     </Appenders>
     <Loggers>
+        <Logger name="javax.mail" level="TRACE" additivity="false">
+            <AppenderRef ref="STDOUT"/>
+        </Logger>
+        <Logger name="org.simplejavamail" level="TRACE" additivity="false">
+            <AppenderRef ref="STDOUT"/>
+        </Logger>
+        <!-- in case you're using authenticated proxy -->
+        <Logger name="socks5bridge" level="TRACE" additivity="false">
```
Comment (andlar) [inline chpl/chpl-api/src/main/resources/log4j2-console.xml]: Are these changes supposed to be in this PR?

File: chpl/chpl-api/src/main/resources/log4j2-console.xml [hunk 5]
```diff
@@ -15,6 +15,19 @@
         </File>
     </Appenders>
     <Loggers>
+        <Logger name="javax.mail" level="TRACE" additivity="false">
```
Comment (kekey1) [inline chpl/chpl-api/src/main/resources/log4j2-console.xml]: Can you remove this?

**Review by kekey1 — CHANGES_REQUESTED** [review]

**Review by andlar — DISMISSED** [review]

**Review by kekey1 — APPROVED** [review]

**Review by andlar — APPROVED** [review]
